src/saveload/saveload.cpp

```diff
@@ -579,6 +580,39 @@ static inline uint SlGetArrayLength(size_t length)
 	return SlGetGammaLength(length);
 }
 
+/**
+ * Return the type as saved/loaded inside the savegame.
+ */
+static uint8 GetSavegameFileType(const SaveLoad &sld)
+{
+	switch (sld.cmd) {
+		case SL_VAR:
+			return GetVarFileType(sld.conv); break;
+
+		case SL_STR:
+		case SL_STDSTR:
+		case SL_ARR:
+		case SL_VECTOR:
+		case SL_DEQUE:
+			return GetVarFileType(sld.conv) | SLE_FILE_HAS_LENGTH_FIELD; break;
+
+		case SL_REF:
+			return IsSavegameVersionBefore(SLV_69) ? SLE_FILE_U16 : SLE_FILE_U32;
```

Comment:
I know refs changed then, though this is only called for save games with a table header, right? In that case the save game version is also 295+, and the check on the save game version is not needed here.

Reply:
Not needed no, but it might be confusing for future-us, as we will forget that this change happened at a later point in time. So I will remove it, and leave a comment to state this. Moving the problem around, basically :D

Reply:
I think this suffers the same fate as my attempts. Future-us has to read it, understand it, and if anything broke around this code, validate it. Which is a lot of actions. With the current code, they check other places using `SL_REF` and go: yeah, this is the same, so very likely to be correct. It is more trivially correct.
But yes, it is strictly seen not correct. In the balance of things, I still prefer what I currently have :) Just because I know how pedantic future-us will be :P
And as you rightfully mention on IRC, they can always find this PR again and this comment. So if we don't resolve this comment, if this ever is a problem, we help future-us again finding back what the consideration was here :D